File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentSessionContextRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentSessionContextModel;

public interface IAgentSessionContextRepository {
    CompletableFuture<AgentSessionContextModel> getAsync(UUID agentId);
    CompletableFuture<Void> upsertAsync(AgentSessionContextModel context);
}

File: apps/backend/src/Features/Agents/Domain/IAgentSessionContextRepository.cs
namespace OffceOs.Features.Agents.Domain;

public interface IAgentSessionContextRepository
{
    Task<AgentSessionContextRecord?> GetByAsync(AgentSessionContextFilter filter, CancellationToken ct = default);
    Task UpsertAsync(AgentSessionContextRecord context, CancellationToken ct = default);
}
